### ISSUE TITLE
cuDNN 5 and Theano 0.9.0 not improving LSTM performance

### ISSUE THREAD

**Issue by mthmn20** [fchollet/keras#4256]
Theano [recently introduced](https://github.com/Theano/Theano/pull/4915) cuDNN5 support for recurrent networks and I decided to give it a spin with a Keras LSTM network. However, I noticed only a marginal benefit that seems more likely noise than anything else. Any reason I shouldn't be seeing bigger performance improvements?

cuDNN 4, theano 0.8.2, keras 1.1.0
Train on 3000000 samples, validate on 857222 samples
Epoch 1/100
3000000/3000000 [==============================] - 781s - loss: 2.2539 - acc: 0.6299 - val_loss: 1.6206 - val_acc: 0.7274

cuDNN 5, theano 0.9.0, keras 1.1.0
Train on 3000000 samples, validate on 857222 samples
Epoch 1/100
3000000/3000000 [==============================] - 765s - loss: 2.2588 - acc: 0.6289 - val_loss: 1.6214 - val_acc: 0.7274

cuDNN 5, theano 0.9.0, keras 1.1.1
Train on 3000000 samples, validate on 857222 samples
Epoch 1/100
3000000/3000000 [==============================] - 761s - loss: 2.2578 - acc: 0.6292 - val_loss: 1.6234 - val_acc: 0.7271

**Comment by nouiz** [comment]
I think this will need new code in keras to get the speed up. There was
another issue related to that.

**Comment by Pyzow** [comment]
Has there been any progress on this or a plan for new keras code :) ?

**Comment by nouiz** [comment]
Not that I know, but I don't follow closely enough keras to be sure.
Hopefully someone else will reply. Otherwise grep in less for gpuarray.dnn
and check if it seem related.
